lib/node/runner/checker_ballot_transaction.go

```diff
@@ -195,16 +197,36 @@ func ValidateOp(st *storage.LevelDBBackend, op transaction.Operation) (err error
 			err = errors.ErrorBlockAccountAlreadyExists
 			return
 		}
+		// If it's a frozen account we check that only whole units are frozen
+		if casted.Linked != "" && (casted.Amount%common.Unit) != 0 {
+			return errors.ErrorFrozenAccountCreationWholeUnit // FIXME
+		}
 	case transaction.OperationPayment:
-		if _, ok := op.B.(transaction.OperationBodyPayment); !ok {
+		var ok bool
+		var casted transaction.OperationBodyPayment
+		if casted, ok = op.B.(transaction.OperationBodyPayment); !ok {
 			err = errors.ErrorTypeOperationBodyNotMatched
 			return
 		}
-		var exists bool
-		if exists, err = block.ExistBlockAccount(st, op.B.(transaction.OperationBodyPayment).Target); err == nil && !exists {
+		var taccount *block.BlockAccount
+		if taccount, err = block.GetBlockAccount(st, casted.Target); err != nil {
 			err = errors.ErrorBlockAccountDoesNotExists
 			return
 		}
+		// If it's a frozen account, it cannot receive payment
+		if taccount.Linked != "" {
+			err = errors.ErrorFrozenAccountNoDeposit
+			return
+		}
+		// If it's a frozen account, everything must be withdrawn
+		if source.Linked != "" {
+			var expected common.Amount
```

Comment:
Where is the code restrict withdrawal from a frozen account within the period?

Reply:
As written in the original post:

> The only thing missing is the actual 2 weeks delay, which is part of #309

Reply:
OK I will hop into that issue.
Thanks. I understand what is going on. @Geod24